dns_check/datadog_checks/dns_check/dns_check.py

```diff
@@ -30,21 +30,15 @@ class BadConfException(Exception):
     pass
 
 
-class DNSCheck(NetworkCheck):
+class DNSCheck(AgentCheck):
     SERVICE_CHECK_NAME = 'dns.can_resolve'
     DEFAULT_TIMEOUT = 5
 
-    def __init__(self, name, init_config, agentConfig, instances=None):
-        # Now that the DNS check is a Network check, we must provide a `name` for each
-        # instance before calling NetworkCheck to make backwards compatible with old yaml.
-        if instances is not None:
-            for idx, inst in enumerate(instances):
-                try:
-                    inst['name'] = inst['name']
-                except KeyError:
-                    inst['name'] = "dns-check-{}".format(idx)
+    def __init__(self, name, init_config, instances):
+        inst = instances[0]
+        inst['name'] = inst.get('name', "dns-check-0")
 
-        NetworkCheck.__init__(self, name, init_config, agentConfig, instances)
+        AgentCheck.__init__(self, name, init_config, instances)
```

Comment:
If you make the above change, can you use super here?

Reply:
yes, I should